src/cache.ts

```diff
@@ -43,3 +47,55 @@ export async function renameCacheFile(): Promise<void> {
 		path.join(appDir(), "old-cache.bak.json"),
 	);
 }
+
+export async function updateTorrentCache(
+	oldUrl: string,
+	newUrl: string,
+): Promise<void> {
+	const torrentFilePaths = await findAllTorrentFilesInDir(
+		path.join(appDir(), TORRENT_CACHE_FOLDER),
+	);
+	console.log(
+		`Found ${torrentFilePaths.length} torrent files in cache, processing...`,
+	);
+	let count = 0;
+	for (const filePath of torrentFilePaths) {
+		try {
+			const torrent: Torrent = bencode.decode(await readFile(filePath));
+			const announce = torrent.announce?.toString();
+			const announceList = torrent["announce-list"]?.map((tier) =>
+				tier.map((url) => url.toString()),
+			);
+			if (
+				announce?.includes(oldUrl) ||
+				announceList?.some((t) => t.some((url) => url.includes(oldUrl)))
+			) {
+				count++;
+				console.log(`#${count}: ${filePath}`);
+				let updated = false;
+				if (announce) {
+					const newAnnounce = announce.replace(oldUrl, newUrl);
```

Comment:
can we call these oldStr and newStr rather than oldUrl and newUrl, given that we support substring replacement?